recipe/passwordless/passwordlessfunctions.go

```diff
@@ -59,3 +62,49 @@ func DefaultCreateAndSendCustomEmail(appInfo supertokens.NormalisedAppinfo) func
 		return nil
 	}
 }
+
+func DefaultCreateAndSendCustomTextMessage(appInfo supertokens.NormalisedAppinfo) func(phoneNumber string, userInputCode *string, urlWithLinkCode *string, codeLifetime uint64, preAuthSessionId string, userContext supertokens.UserContext) error {
+	return func(phoneNumber string, userInputCode *string, urlWithLinkCode *string, codeLifetime uint64, preAuthSessionId string, userContext supertokens.UserContext) error {
+		if supertokens.IsRunningInTestMode() {
+			// if running in test mode, we do not want to send this.
+			PasswordlessLoginSmsSentForTest = true
+			return nil
+		}
+
+		data := map[string]map[string]interface{}{
+			"smsInput": {
+				"type":         "PASSWORDLESS_LOGIN",
+				"phoneNumber":  phoneNumber,
+				"codeLifetime": codeLifetime,
+				"appName":      appInfo.AppName,
+			},
+		}
+		if urlWithLinkCode != nil {
+			data["smsInput"]["urlWithLinkCode"] = *urlWithLinkCode
+		}
+		if userInputCode != nil {
+			data["smsInput"]["userInputCode"] = *userInputCode
+		}
+
+		jsonData, err := json.Marshal(data)
+		if err != nil {
+			return err
+		}
+		req, err := http.NewRequest("POST", supertokensService.SUPERTOKENS_SMS_SERVICE_URL, bytes.NewBuffer(jsonData))
+		if err != nil {
+			return err
+		}
+
+		req.Header.Set("content-type", "application/json")
+		req.Header.Set("api-version", "0")
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode == 429 {
+			return errors.New("Free daily SMS quota reached. If using our managed service, please create a production environment to get dedicated API keys for SMS sending, or define your own method for sending SMS.")
```

Comment:
This is not an error state. We are to log this and we are to log the SMS content (input of the function)

Reply:
I have updated this part. I will review all the API calls once again and raise a separate PR for that.